File: 0001-two-sum/0001-two-sum.java


class Solution {
    public int[] twoSum(int[] nums, int target) {
        
        HashMap<Integer,Integer> hash = new HashMap<>();

        for (int i = 0; i < nums.length; i++){
            if (hash.containsKey(target - nums[i])){
                int toReturn[] = {hash.get(target - nums[i]),i};
                return toReturn;
            }
            hash.put(nums[i],i);
        }

        return new int[] {};
    }
}

File: 0001-two-sum/0001-two-sum.cpp
#include<stdio.h>
#include <map>
#include<iostream>    
#include<vector> 
using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int, int> missing;
        std::vector<int> myvec;

        for (int i = 0; i < nums.size(); i++){
            if (missing.find(target - nums[i]) == missing.end()){
                 missing[nums[i]] = i;
            }
            else
                return {missing[target - nums[i]], i};
        }

        return {-1,-1};
    }
};